runtime-tokio/src/lib.rs

```diff
@@ -81,3 +83,70 @@ impl runtime_raw::Runtime for Tokio {
         Ok(Box::pin(UdpSocket { tokio_socket }))
     }
 }
+
+/// The single-threaded Tokio runtime based on `tokio-current-thread`.
+#[derive(Debug)]
+pub struct TokioCurrentThread;
+
+impl runtime_raw::Runtime for TokioCurrentThread {
+    fn spawn_obj(&self, fut: FutureObj<'static, ()>) -> Result<(), SpawnError> {
+        lazy_static! {
+            static ref TOKIO_RUNTIME: Mutex<tokio::runtime::current_thread::Handle> = {
+                let (tx, rx) = mpsc::channel();
```

Comment:
this is a bit unfortunate 😢 , the only way around this I think is to create your own runtime and decouple the running of it with creating the handle. Within the tokio-current-thread crate you can manually turn and do all that. But prob not worth it.

Reply:
Yeah, agree on all fronts. Perhaps someday we can find a way to do that, but hopefully for now this should be good enough (: